include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1480,74 +1483,92 @@ struct __parallel_find_or_impl_one_wg<__or_tag_check, __internal::__optional_ker
     }
 };
 
-template <bool __or_tag_check, typename KernelName>
+template <bool __or_tag_check, typename KernelNameInit, typename KernelName>
 struct __parallel_find_or_impl_multiple_wgs;
 
 // Base pattern for __parallel_or and __parallel_find. The execution depends on tag type _BrickTag.
-template <bool __or_tag_check, typename... KernelName>
-struct __parallel_find_or_impl_multiple_wgs<__or_tag_check, __internal::__optional_kernel_name<KernelName...>>
+template <bool __or_tag_check, typename... KernelNameInit, typename... KernelName>
+struct __parallel_find_or_impl_multiple_wgs<__or_tag_check, __internal::__optional_kernel_name<KernelNameInit...>,
+                                            __internal::__optional_kernel_name<KernelName...>>
 {
     template <typename _BrickTag, typename _AtomicType, typename _Predicate, typename... _Ranges>
     _AtomicType
     operator()(sycl::queue& __q, _BrickTag __brick_tag, const std::size_t __rng_n, const std::size_t __n_groups,
                const std::size_t __wgroup_size, const _AtomicType __init_value, _Predicate __pred, _Ranges&&... __rngs)
     {
-        auto __result = __init_value;
+        // Allocate memory for result in USM device memory, if it's available:
+        // - we can't use USM host or USM shared memory for atomic operations through __dpl_sycl::__atomic_ref.
```

Comment:
https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2024-0/atomic-operations.html

Atomic Operations on USM Data
On discrete GPU,

Atomic operations on host allocated USM (sycl::malloc_host) are not supported.

Concurrent accesses from host and device to shared USM location (sycl::malloc_shared) are not supported.

We recommend using device allocated USM (sycl::malloc_device) memory for atomics and device algorithms with atomic operations.